### PR TITLE
feat/refactor: add constant pkg

### PR DESCRIPTION
**Pull request by yqlbu**
### Background

Add `constant` pkg

New structure:

```console
◆ juicity git:(refactor_config) ❯❯❯ rg --files common/ --files internal/ --files server/ | tree --fromfile                                                                 19:00:19
.
├── common
│   └── consts
│       └── relay.go
├── internal
│   └── relay
│       ├── relay.go
│       ├── tcp.go
│       └── udp.go
└── server
    ├── mixed_server.go
    └── server.go

6 directories, 6 files
```

### Checklist

- [x] The Pull Request has been fully tested
- [ ] There's an entry in the CHANGELOGS
- [ ] There is a user-facing docs PR against https://github.com/daeuniverse/dae

### Full changelogs

- refactor: relocate /config -> /internal/config
- refactor(constant): import relay constants from /internal/constant

### Issue reference

NA

### Test Result

CleanShot 2023-07-30 at 15 45 47@2x

### CODE REVIEW

**Comment by mzz2017** [comment]
Do not move config. It is a project-related package. Outer package should use it to create servers.

**Comment by yqlbu** [comment]
> Do not move config. It is a project-related package. Outer package should use it to create servers.

Ok. Thanks for the clarification. How about adding the `constant` package? e.g constants for relay go to`/internal/constant/relay.go`; constants for log go to `/internal/constant/log.go`

**Comment by yqlbu** [comment]
> > Do not move config. It is a project-related package. Outer package should use it to create servers.
> 
> Ok. Thanks for the clarification. How about adding the `constant` package? e.g constants for relay go to`/internal/constant/relay.go`; constants for log go to `/internal/constant/log.go`

Resolved in https://github.com/juicity/juicity/pull/31/commits/40500e02634a4967d4a37216f3089fee6e39e36e

File: internal/relay/relay.go [hunk 1]
```diff
@@ -33,5 +30,10 @@ type WriteCloser interface {
 }
 
 func NewRelay(logger *log.Logger) Relay {
-	return &relay{logger: logger}
+	return &relay{
+		logger:          logger,
+		mtu:             constant.EthernetMtu,
```
Comment (mzz2017) [inline internal/relay/relay.go:35]: I don't think we should duplicate constants as members. It is meaningless.

File: internal/constant/relay.go [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package constant
```
Comment (mzz2017) [inline internal/constant/relay.go:1]: It is better to be shorter. What about consts?

Comment (yqlbu) [reply internal/constant/relay.go]: Good idea.

Comment (yqlbu) [reply internal/constant/relay.go]: Resolved in https://github.com/juicity/juicity/pull/31/commits/2cbe7ac8f0dcd7c7680c3746f3d0be846f83ea7b

File: internal/relay/relay.go (hunk 1 above)
Comment (yqlbu) [reply internal/relay/relay.go]: Sure, good catch. Resolved in https://github.com/juicity/juicity/pull/31/commits/664ed3cbc527abe035d60aaa50e86f1852bae0a5

File: go.mod [hunk 3]
```diff
@@ -5,8 +5,8 @@ go 1.20
 require (
 	github.com/google/uuid v1.3.0
 	github.com/miekg/dns v1.1.55
-	github.com/mzz2017/quic-go v0.0.0-20230730112514-5fa9b68b3e17
-	github.com/mzz2017/softwind v0.0.0-20230730112720-7959c1d1482c
+	github.com/mzz2017/quic-go v0.0.0-20230730161358-da6d530ddb18
```
Comment (mzz2017) [inline go.mod:8]: LGTM. But do we need to revert this commit before merging?

Comment (mzz2017) [reply go.mod:8]: https://github.com/juicity/juicity/commit/70643fc4d2ee66f3410c8dac2c28fdafdc03ffa5
@yqlbu

Comment (yqlbu) [reply go.mod]: Sure, https://github.com/juicity/juicity/pull/31/commits/a688378a8e47d561133a2ae60fb7485742c4cbac

CleanShot 2023-07-31 at 20 04 21@2x

**Review by mzz2017 — APPROVED** [review]

**Comment by mzz2017** [comment]
LGTM

**Pull request merged**